00_homework_kawapa/Z4.cpp

```diff
@@ -15,11 +16,12 @@ struct Philosopher
     int id_;
     int forkIndexLeft_;
     int forkIndexRight_;
-    bool hungry_ = true;
+    int eated_ = 0;
+    std::vector<Philosopher> & v_;
```

Comment:
Dajesz filozofom dostęp do wszystkich innych. Przez to w teorii możesz na dowolnym innym filozofie wywoływać jego funkcje publiczne albo modyfikować co Ci się podoba. To łamie enkapsulację.
Lepiej będzie, jak filozof będzie znał tylko swoich sąsiadów, a jeszcze lepiej, jak to ktoś z zewnątrz (tzw. kelner, arbiter) będzie widział kiedy kto ostatnio jadł i on będzie decydował kto ma pierwszeństwo do widelców.